packages/fuel-gauge/src/predicate/predicate-estimations.test.ts

```diff
@@ -74,14 +79,48 @@ describe('Predicate', () => {
         predicateGasUsed: bn(0),
         predicateData: '0x',
       };
-      tx.inputs.push(inputMessage);
+
+      const nonPredicateMessage: MessageTransactionRequestInput = {
+        type: InputType.Message,
+        amount: bn(100),
+        sender: getRandomB256(),
+        recipient: getRandomB256(),
+        witnessIndex: 0,
+        data: '0x',
+        nonce: getRandomB256(),
+      };
+
+      const contract: ContractTransactionRequestInput = {
+        type: 1,
+        contractId: '0xb64fa600c9a940529020fd47a18f9c1395946fbf636aad13fe029db4820ed354',
+        txPointer: '0x00000000000000000000000000000000',
+      };
+
+      const nonPredicateUtxo: CoinTransactionRequestInput = {
+        id: '0x5b2f4599a29aea28e325c89249c9e397f8b86bbd2405cf3b0face56c8e0e4dbe01',
+        amount: bn(100),
+        assetId: '0x0000000000000000000000000000000000000000000000000000000000000000',
+        owner: '0xd8813d1f9ca165ce2e8710382c3d65d64e7bd43c0f7a3d51689bcdf9513411cd',
+        maturity: 0,
+        type: 0,
+        txPointer: '0x00000000000000000000000000000000',
+        witnessIndex: 0,
+      };
+
+      tx.inputs.push(predicateMessage, nonPredicateMessage, contract, nonPredicateUtxo);
 
       const txEstimated = await provider.estimatePredicates(tx);
 
       const predicateCoinInput = <MessageTransactionRequestInput>txEstimated.inputs[0];
       expect(Number(predicateCoinInput.predicateGasUsed)).toBeGreaterThan(1);
       const predicateMessageInput = <MessageTransactionRequestInput>txEstimated.inputs[1];
       expect(Number(predicateMessageInput.predicateGasUsed)).toBeGreaterThan(1);
+      const nonPredicateMessageInput = <MessageTransactionRequestInput>txEstimated.inputs[2];
+      expect(nonPredicateMessageInput.predicateGasUsed).toBeUndefined();
+      const contractInput = <ContractTransactionRequestInput>txEstimated.inputs[3];
+      expect(contractInput.contractId).toBe(contract.contractId);
+      const nonPredicateInput = <CoinTransactionRequestInput>txEstimated.inputs[4];
+      expect(nonPredicateInput.predicateGasUsed).toBeUndefined();
```

Comment:
I haven't reviewed this yet, but I want to point out the same concerns I posted [here](https://github.com/FuelLabs/fuels-ts/pull/1710#pullrequestreview-1862291286).

In short, when we change GQL fragments, we need to be sure that all underlying related types and method utilities are updated accordingly, as selectively hard-casting things can result in incomplete data being shuffled around.